File: python/linearSearchRecursive.py
#   Linear Search Recursive Algorithm in Python
#
#   Author: Elton Fonseca

def main():

    array = [1,2,3,4,5,6,7,8,9,10]

    if search(array, len(array) - 1, 6):
        print "Find!"
    else:
        print "Not Found!"

def search(array, element, value):
    if element < 0:
        return 0
    elif array[element] == value:
        return 1
    else:
        return search(array, element - 1, value)

main()

File: ruby/linearSearchRecursive.rb
#   Linear Search Recursive Algorithm in Ruby
#
#   Author: Elton Fonseca
#

def main()
    @array = [1,2,3,4,5,6,7,8,9,10]

    if(search(@array, @array.length - 1, 7))
        puts("Find!")
    else
        puts("Not Found!")
    end 
end

def search(array, element, value)
    if(element < 0)
        return false
    elsif(array[element] == value)
        return true
    else
        return search(array, element - 1, value)
    end
end

main()